### ISSUE TITLE
Crash `AsSuperVisitor: type is not an erased subtype of supertype` on chained method references over a wildcard return type

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#2098]
The Nullness Checker crashes on this code. javac alone compiles it.

```java
import java.util.stream.Stream;

class G {
    interface P<T> {
        Class<? extends T> type();
    }

    Stream<String> names(Stream<P<Object>> providers) {
        return providers.map(P::type).map(Class::getName);
    }
}
```

```
javac -processor nullness G.java
```

```
error: AsSuperVisitor: type is not an erased subtype of supertype.
  type: Object
  superType: Class<Object>
  ; The Checker Framework crashed.  Please report the crash.
  Checker: class org.checkerframework.checker.nullness.KeyForSubchecker
  	at org.checkerframework.framework.type.AsSuperVisitor.errorTypeNotErasedSubtypeOfSuperType(AsSuperVisitor.java:152)
```

The same crash is also reported by `InitializationFieldAccessSubchecker` and `NullnessChecker`.

| Variant | Result |
|---|---|
| as above | crash |
| `Class<T> type()` instead of `Class<? extends T>` | no crash |
| `.map(p -> p.type()).map(c -> c.getName())` | no crash |
| only `.map(P::type)` | no crash |
| `Class` replaced by a user-defined `interface Box<X> { String name(); }` | crash |
| `P<String>` instead of `P<Object>` | crash |
| `-AassumeInitialized -AassumeKeyFor` | no crash |
| `-AonlyAnnotatedFor` | crash |

Found in junit-framework, `junit-platform-commons/.../ClasspathScannerLoader.java:39`: `classpathScanners.stream().map(Provider::type).map(Class::getName).toList()`.

Environment: EISOP 3.49.5-eisop2-SNAPSHOT (master, 7e42d55), javac 25.

**Comment by aosen-xiong** [comment]
pull in upstream's 8034 and 8083 will fix this problem.
